### PR TITLE
[EN TasksMode] Merge meal time references with date, if contiguous in text

### PR DESCRIPTION
**Pull request by Neelisha-saxena**
Example: "Take medicine after dinner tomorrow" 
Under Default mode, "after dinner" and "tomorrow" are separate recognition.  But under TasksMode "after dinner tomorrow" is a single recognition.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -221,6 +221,7 @@ public static class DateTimeDefinitions
       public static readonly string PeriodTimeOfDayRegex = $@"\b((in\s+(the)?\s+)?{LaterEarlyRegex}?((this\s+)?{DateTimeTimeOfDayRegex}|(?<timeOfDay>(?<pm>tonight))))\b";
       public static readonly string PeriodSpecificTimeOfDayRegex = $@"\b({LaterEarlyRegex}?this\s+{DateTimeTimeOfDayRegex}|({StrictRelativeRegex}\s+{PeriodTimeOfDayRegex})\b|\b(?<pm>toni(ght|te)))\b";
       public static readonly string PeriodTimeOfDayWithDateRegex = $@"\b(({PeriodTimeOfDayRegex}(\s+(on|of))?))\b";
+      public static readonly string TasksmodeMealTimeofDay = $@"\b((in\s+(the)?\s+)?((?<early>earl(y|ier)(\s+|-))|(?<late>late(r?\s+|-)))?((this\s+)?\b(?<timeOfDay>lunch(\s*time)?|dinner(\s*time)?|brunch|breakfast)\b))\b";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:224]: This seems too brittle for merge. I'm OK with it to streamline merging your new mode behaviour. But I'd suggest you look at some of the PRs by @aitelint to see how merge was done in other cases.
These specific merges could potentially be applied to the default mode (pending also perf).

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs [hunk 2]
```diff
@@ -64,7 +65,7 @@ public EnglishDateTimePeriodParserConfiguration(ICommonDateTimeParserConfigurati
             PrefixDayRegex = EnglishDateTimePeriodExtractorConfiguration.PrefixDayRegex;
             BeforeRegex = EnglishDateTimePeriodExtractorConfiguration.BeforeRegex;
             AfterRegex = EnglishDateTimePeriodExtractorConfiguration.AfterRegex;
-
+            TasksmodeMealTimeofDay = EnglishDateTimePeriodExtractorConfiguration.TasksmodeMealTimeofDay;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs:68]: Nitpick: Mode this init to the last and add a blank line before it.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs [hunk 3]
```diff
@@ -148,6 +151,8 @@ public EnglishDateTimePeriodExtractorConfiguration(IDateTimeOptionsConfiguration
 
         Regex IDateTimePeriodExtractorConfiguration.PeriodTimeOfDayWithDateRegex => PeriodTimeOfDayWithDateRegex;
 
+        Regex IDateTimePeriodExtractorConfiguration.TasksmodeMealTimeofDay => TasksmodeMealTimeofDay;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs:154]: TasksmodeMealTimeofDayRegex

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs [hunk 4]
```diff
@@ -60,6 +60,8 @@ public interface IDateTimePeriodExtractorConfiguration : IDateTimeOptionsConfigu
 
         Regex AfterRegex { get; }
 
+        Regex TasksmodeMealTimeofDay { get; }
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs:63]: TasksmodeMealTimeofDayRegex

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+        // Change resolution value of datetime value under tasksmode.
+        public static TimeOfDayResolutionResult TasksModeResolveTimeOfDay(string tod)
+        {
+            var result = new TimeOfDayResolutionResult();
+            switch (tod)
+            {
+                case Constants.EarlyMorning:
+                    result.Timex = Constants.EarlyMorning;
+                    result.BeginHour = 6;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs:22]: Same as in previous PR.

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs (hunk 5 above)
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs]: added tasksmode constants.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs (hunk 2 above)
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs:68]: changes integrated

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs (hunk 3 above)
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs]: renamed variable

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs (hunk 4 above)
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs]: renamed var

File: .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+using System.Globalization;
+
+using Microsoft.Recognizers.Definitions;
+
+namespace Microsoft.Recognizers.Text.DateTime
+{
+    [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1310: CSharp.Naming : Field names must not contain underscores.", Justification = "Constant names are written in upper case so they can be readily distinguished from camel case variable names.")]
+    public static class TasksModeConstants
+    {
+        public const string SYS_DATETIME_DATE = "date";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs:14]: Same comment as in previous PR.

**Review by tellarin — CHANGES_REQUESTED** [review]
This PR itself LGTM, but pending a fix from another PR before it can be rebased and merged.

File: .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs (hunk 6 above)
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs:14]: resolved similar to previous PR.

**Review by tellarin — APPROVED** [review]

**Comment by tellarin** [comment]
@Neelisha-saxena Please resolve the conflicts to merge this one first. Thanks.

**Comment by Neelisha-saxena** [comment]
> @Neelisha-saxena Please resolve the conflicts to merge this one first. Thanks.

resolved merge conflicts.

**Review by tellarin — APPROVED** [review]

**Pull request merged**